src/KMonad/App/Main.hs

```diff
@@ -84,16 +84,19 @@ initAppEnv cfg = do
   src <- using $ cfg^.keySourceDev
 
   -- Initialize the pull-chain components
-  dsp <- Dp.mkDispatch $ awaitKey src
-  ihk <- Hs.mkHooks    $ Dp.pull  dsp
-  slc <- Sl.mkSluice   $ Hs.pull  ihk
+  itr <- atomically $ newEmptyTMVar
+  dsp <- Dp.mkDispatch (awaitKey src) itr
+  ihkPrio <- Hs.mkHooks (Dp.pull dsp) itr
+  slc <- Sl.mkSluice $ Hs.pull ihkPrio
+  ihk <- Hs.mkHooks (Sl.pull slc) itr
 
   -- Initialize the button environments in the keymap
   phl <- Km.mkKeymap (cfg^.firstLayer) (cfg^.keymapCfg)
 
   -- Initialize output components
   otv <- lift . atomically $ newEmptyTMVar
-  ohk <- Hs.mkHooks . atomically . takeTMVar $ otv
+  otr <- atomically $ newEmptyTMVar
+  ohk <- Hs.mkHooks ((atomically . takeTMVar) otv >>= pure . (WrappedKeyEvent NoCatch)) otr
```

Comment:
When changing the parameter order of `mkHooks` as described above this too would be more readable\*:
```suggestion
  ohk <- Hs.mkHooks otr $ pure . WrappedKeyEvent NoCatch <=< atomically . takeTMVar $ otv
```
\* readablitiy is subjective so this is just my opinion.